driver-async/src/test/unit/com/mongodb/async/client/SingleResultCallbackSubscriptionSpecification.groovy

```diff
@@ -289,6 +289,25 @@ class SingleResultCallbackSubscriptionSpecification extends Specification {
         observer.assertErrored()
     }
 
+    def 'should call onError on a null result'() {
+        given:
+        def observer = new TestObserver()
+        observe(new Block<SingleResultCallback<Void>>() {
+            @Override
+            void apply(final SingleResultCallback<Void> callback) {
+                callback.onResult(null, null);
+            }
+        }).subscribe(observer)
+
+        when:
+        observer.requestMore(1)
+
+        then:
+        notThrown(MongoException)
+        observer.assertTerminalEvent()
```

Comment:
Can we assert that it was a MongoClientException that was passed to onError??

Reply:
Done